src/authorization/jwt/JwtValidator.ts

```diff
@@ -4,55 +4,69 @@
 import * as ccfapp from "@microsoft/ccf-app";
 import { ServiceResult } from "../../utils/ServiceResult";
 import { IValidatorService } from "../IValidationService";
-import { JwtIdentityProviderEnum } from "./JwtIdentityProviderEnum";
-import { IJwtIdentityProvider } from "./IJwtIdentityProvider";
-import { DemoJwtProvider } from "./DemoJwtProvider";
-import { MsJwtProvider } from "./MsJwtProvider";
 import { Logger, LogContext } from "../../utils/Logger";
+import { JwtValidationPolicyMap } from "./JwtValidationPolicyMap";
+import isMatch from 'lodash/isMatch';
 
 export class JwtValidator implements IValidatorService {
-  private readonly identityProviders = new Map<
-    JwtIdentityProviderEnum,
-    IJwtIdentityProvider
-  >();
   private logContext: LogContext;
 
   constructor(logContext?: LogContext) {
     this.logContext = (logContext?.clone() || new LogContext()).appendScope("JwtValidator");
-    this.identityProviders.set(
-      JwtIdentityProviderEnum.MS_AAD,
-      new MsJwtProvider("JwtProvider", this.logContext),
-    );
-    this.identityProviders.set(
-      JwtIdentityProviderEnum.Demo,
-      new DemoJwtProvider("DemoJwtProvider"),
-    );
   }
 
   validate(request: ccfapp.Request<any>): ServiceResult<string> {
     const jwtCaller = request.caller as unknown as ccfapp.JwtAuthnIdentity;
     Logger.debug(
-      `Authorization: JWT jwtCaller (JwtValidator)-> ${<JwtIdentityProviderEnum>jwtCaller.jwt.keyIssuer}`,
+      `Authorization: JWT jwtCaller (JwtValidator)-> ${jwtCaller.jwt.keyIssuer}`,
       this.logContext
     );
-    const provider = this.identityProviders.get(
-      <JwtIdentityProviderEnum>jwtCaller.jwt.keyIssuer,
+    const issuer = jwtCaller?.jwt?.payload?.iss;
+    if (!issuer) {
+      return ServiceResult.Failed(
+        {
+          errorMessage: "The JWT has no valid iss",
+          errorType: "AuthenticationError",
+        },
+        400,
+        this.logContext
+      );
+    }
+
+
+    const policy = JwtValidationPolicyMap.read(issuer, this.logContext);
+    if (policy === undefined) {
+      const errorMessage = `issuer ${issuer} is not defined in the policy`;
+      Logger.error(errorMessage, this.logContext);
+      return ServiceResult.Failed(
+        {
+          errorMessage,
+          errorType: "AuthenticationError",
+        },
+        500,
+        this.logContext
+      );
+    }
+    Logger.debug(
+      `Validate JWT policy for issuer ${issuer}: ${JSON.stringify(policy)}`, this.logContext
     );
 
-    if (!provider) {
-      const error = `Authorization: JWT validation provider is undefined (JwtValidator) for ${<JwtIdentityProviderEnum>jwtCaller.jwt.keyIssuer}`;
-      Logger.error(error, this.logContext);
+    // Check that the JWT payload is a matching superset of the policy
+    if (!isMatch(jwtCaller.jwt.payload, policy)) {
```

Comment:
We need to be able to support array values for token properties, we should go back to previous

Reply:
Done in https://github.com/microsoft/azure-privacy-sandbox-kms/pull/364/commits/2a703a431ce4327305b887af265ee972f273938b